setup.py

```diff
@@ -89,7 +89,12 @@ def run(self):
 azure = ["azure-storage-blob==2.1.0"]
 oss = ["oss2==2.6.1"]
 ssh = ["paramiko>=2.5.0"]
-hdfs = ["pyarrow==0.15.1"]
+hdfs = [
+    # pyarrow-0.16.0 import fails on 3.5 and 3.7 (works on 3.6 though)
+    # due to: https://issues.apache.org/jira/browse/ARROW-7852
+    "pyarrow==0.15.1; python_version < '3.8'",
+    "pyarrow==0.16.0; python_version == '3.8'",
```

Comment:
```suggestion
    "pyarrow==0.16.0; python_version == '3.8' or python_version == '3.6'",
```

Reply:
@casperdcl, I'd love to leave 3.6 on older version for now due to complications in `pyarrow` itself. Maybe, when we get to `0.17.0` of pyarrow, we could update on all python versions.